src/Microsoft.AspNet.Mvc.TagHelpers/EnvironmentTagHelper.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting &lt;environment&gt; elements that conditionally renders content
+    /// based on the current value of <see cref="IHostingEnvironment.EnvironmentName"/>.
+    /// </summary>
+    public class EnvironmentTagHelper : TagHelper
+    {
+        private static readonly char[] _nameSeparator = new[] { ',' };
+
+        /// <summary>
+        /// A comma separated list of environment names in which the content should be rendered.
+        /// </summary>
+        public string Names { get; set; }
+
+        // Protected to ensure subclasses are correctly activated. Internal for ease of use when testing.
+        [Activate]
+        protected internal IHostingEnvironment HostingEnvironment { get; set; }
+
+        /// <inheritdoc />
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            // Always strip the outer tag name as we never want <environment> to render
+            output.TagName = null;
```

Comment:
I thought about that myself but decided against it given the extra complexity it adds and we'd have to do it everywhere we create custom elements like this. Web Forms didn't clear this whitespace either. I used to work around it on pages I *really* cared about by making the whitespace commented out, e.g.:

``` HTML
<environment names="Development">@*
    *@This is my content@*
*@</environment>
```